src/staticanalysis/bot/static_analysis_bot/clang/tidy.py

```diff
@@ -244,6 +244,11 @@ def __init__(self, header_data, revision):
         self.body = None
         self.notes = []
 
+        self.reason = None
+        for c in settings.clang_checkers:
```

Comment:
This part will not work in every case, you need to support wildcard matches (for example we use `mozilla-*` to describe all checks which name start with `mozilla-`.

Please look [this code](https://github.com/mozilla/release-services/blob/4625854c25aa6630184751a85325d30de2756c8d/src/staticanalysis/bot/static_analysis_bot/config.py#L106), and add a generic function such that `settings.get_check(check_name)` returns a check. You can then use it here and in the `is_publishable_check` function.